src/node/controller/messages.rs

```diff
@@ -136,9 +137,53 @@ impl Controller {
     pub async fn handle_publish(
         &mut self,
         _header: &Header,
-        _publish: Publish,
+        publish: Publish,
     ) -> anyhow::Result<()> {
         // dbg!(publish);
+        let _block = match publish.block_holder {
+            BlockHolder::Send(_) => {
+                todo!("Received a send block")
+            }
+            BlockHolder::Receive(_) => {
+                todo!("Received a receive block")
+            }
+            BlockHolder::Open(_) => {
+                todo!("Received an open block")
+            }
+            BlockHolder::Change(_) => {
+                todo!("Received a change block")
+            }
+            BlockHolder::State(state_block) => {
+                let mut block = Block::from_state_block(&state_block);
+                let block_hash: &BlockHash = {
+                    block.calc_hash().unwrap();
+                    block.hash().unwrap()
+                };
+                // dbg!(state_block);
+
+                // # deduplication
```

Comment:
What's the `#` for? 😅

Reply:
I means important comment here! Just kidding, I'm going to delete all the comments before committing anyway.